src/www/services_dhcp_edit.php

```diff
@@ -493,6 +493,24 @@ function show_tftp_config() {
                       <input name="ddnsdomainkeyname" type="text" id="ddnsdomainkeyname" size="20" value="<?=$pconfig['ddnsdomainkeyname'];?>" />
                       <?=gettext("Enter the dynamic DNS domain key secret which will be used to register client names in the DNS server.");?>
                       <input name="ddnsdomainkey" type="text" id="ddnsdomainkey" size="20" value="<?=$pconfig['ddnsdomainkey'];?>" />
+                      <?=gettext("Choose the dynamic DNS domain key algorithm.");?><br />
+                      <select name='ddnsdomainalgorithm' id="ddnsdomainalgorithm" class="selectpicker">
+<?php
+                        foreach (array("hmac-md5", "hmac-sha512") as $algorithm) :
+                          $selected = "";
+                          if (! empty($pconfig['ddnsdomainalgorithm'])) {
+                            if ($pconfig['ddnsdomainalgorithm'] == $algorithm) {
+                              $selected = "selected=\"selected\"";
+                            }
+                          } else {
+                            if ($algorithm == "hmac-md5") {
+                              $selected = "selected=\"selected\"";
```

Comment:
First algo in the list is selected by default by the browser which makes this superfluous.